cmd/cmd.go

```diff
@@ -0,0 +1,176 @@
+// Package cmd - package
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github/Asac2142/go-cli-app/file"
+	"github/Asac2142/go-cli-app/task"
+)
+
+const (
+	add            string = "add"
+	update         string = "update"
+	deleteTask     string = "delete"
+	markInProgress string = "mark-in-progress"
+	markDone       string = "mark-done"
+	list           string = "list"
+)
+
+// HandleTrackerCLI - Handles user's input
+func HandleTrackerCLI(f *file.File[task.TContent]) {
+	task := task.NewTask(f)
+	args := os.Args[1:]
+
+	if len(args) == 0 {
+		err := errors.New("Invalid input")
+		printError(err)
+	}
+
+	switch args[0] {
+	case add:
+		err := addTask(&args, task)
+		printError(err)
```

Comment:
suppose you already replaced it with return wrapped error. Now as you can see, all the switch branches have the same error handling, so you can more it out of the switch.

So you need err to be declared before switch, in each case to assign command error to err variable and after the switch you do
```
if err != nil {
    return fmt.Errorf("%v: %w", args[0], err)
}
```

Reed docs about fmt.Errorf and use it EVERY time you propagate error adding meaningful human readable context.